app/src/main/java/org/simple/clinic/di/network/CompressRequestInterceptor.kt

```diff
@@ -0,0 +1,30 @@
+package org.simple.clinic.di.network
+
+import okhttp3.Interceptor
+import okhttp3.Request
+import okhttp3.Response
+import javax.inject.Inject
+
+/**
+ * Class that supports GZIP encoding for HTTP request bodies.
+ *
+ * Copied from https://github.com/square/okhttp/blob/8bb58332dbacdd2bb9beb8e5c813f22508ebeb27/samples/guide/src/main/java/okhttp3/recipes/RequestBodyCompression.java
+ **/
+class CompressRequestInterceptor @Inject constructor(
+    private val requestCompression: RequestCompression
+) : Interceptor {
+
+  override fun intercept(chain: Interceptor.Chain): Response {
+    val originalRequest = chain.request()
+
```

Comment:
We have not added any tests for this class and we have an empty test class for this. Is there a reason for this?

Reply:
Ah, good catch. I wanted to test the interceptor, but that required a bit of setup since it was dependent on some OkHttp specific components, which was irrelevant to this PR. So I instead used the `RequestCompression` to implement the relevant bits and tested that instead. But I think I forgot to delete the empty test file I created for testing the interceptor.

Will delete this.

Reply:
Done.